source/Octopus.Tentacle/Commands/RegisterMachineCommandBase.cs

```diff
@@ -79,11 +82,19 @@ public abstract class RegisterMachineCommandBase<TRegistrationOperationType> : A
 
         protected override void Start()
         {
+            cancellationTokenSource = new CancellationTokenSource();
```

Comment:
Feels like the cancellation should move up to potentially the `OctopusProgram` level to be passed down and handled at that level rather at the command level